src/index/index.cc

```diff
@@ -34,6 +36,40 @@ LoadConfig(BaseConfig* cfg, const Json& json, knowhere::PARAM_TYPE param_type, c
     return Config::Load(*cfg, json_, param_type, msg);
 }
 
+#ifdef KNOWHERE_WITH_CARDINAL
+template <typename T>
+inline const std::unique_ptr<Handle>
+Index<T>::BuildAsync(const DataSetPtr dataset, const Json& json, const std::chrono::seconds timeout) {
+    auto pool = ThreadPool::GetGlobalBuildThreadPool();
+    auto handle = std::make_unique<Handle>(timeout);
```

Comment:
Please switch to `std::shared_ptr<Handle>` here. The following `handle->Set()` line does not convince me that we won't have a future lifetime problems, and I'm not sure whether Rust compiler would easily allow this form of self-reference, for example. Making `pool->push()` capture `std::shared_ptr` (not `std::shared_ptr&`) seems more safe.

Reply:
Alex, you are correct. There's a potential risk here. If the code that calls `buildAsync` directly discards the return value, the thread actually executing the build might encounter illegal access issues.